src/test/regress/expected/background_rebalance_parallel.out

```diff
@@ -583,6 +586,257 @@ SELECT citus_rebalance_stop();
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO table1_colg5 SELECT i FROM generate_series(0, 100)i;
+CREATE TABLE table1_colg6 (a int PRIMARY KEY);
+SELECT create_distributed_table('table1_colg6', 'a', shard_count => 1, colocate_with => 'none');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO table1_colg6 SELECT i FROM generate_series(0, 100)i;
+-- add two other nodes
+-- nodeid here is 59
+select citus_add_node('localhost', :worker_2_port);
+ citus_add_node
+---------------------------------------------------------------------
+             59
+(1 row)
+
+-- nodeid here is 60
+select citus_add_node('localhost', :worker_3_port);
+ citus_add_node
+---------------------------------------------------------------------
+             60
+(1 row)
+
+CREATE OR REPLACE FUNCTION shard_placement_rebalance_array(
+    worker_node_list json[],
+    shard_placement_list json[],
+    threshold float4 DEFAULT 0,
+    max_shard_moves int DEFAULT 1000000,
+    drain_only bool DEFAULT false,
+    improvement_threshold float4 DEFAULT 0.5
+)
+RETURNS json[]
+AS 'citus'
+LANGUAGE C STRICT VOLATILE;
+-- we are simulating the following from shard_rebalancer_unit.sql
+-- the following steps are all according to this scenario
+-- where the third move should be dependent of the first two
+-- because the third move's target is the source of the first two
+SELECT unnest(shard_placement_rebalance_array(
+    ARRAY['{"node_name": "hostname1", "disallowed_shards": "1,2,3,5,6"}',
+          '{"node_name": "hostname2", "disallowed_shards": "4"}',
+          '{"node_name": "hostname3", "disallowed_shards": "4"}'
+        ]::json[],
+    ARRAY['{"shardid":1, "nodename":"hostname1"}',
+          '{"shardid":2, "nodename":"hostname1"}',
+          '{"shardid":3, "nodename":"hostname2"}',
+          '{"shardid":4, "nodename":"hostname2"}',
+          '{"shardid":5, "nodename":"hostname3"}',
+          '{"shardid":6, "nodename":"hostname3"}'
+        ]::json[]
+));
+                                                       unnest
+---------------------------------------------------------------------
+ {"updatetype":1,"shardid":1,"sourcename":"hostname1","sourceport":5432,"targetname":"hostname2","targetport":5432}
+ {"updatetype":1,"shardid":2,"sourcename":"hostname1","sourceport":5432,"targetname":"hostname3","targetport":5432}
+ {"updatetype":1,"shardid":4,"sourcename":"hostname2","sourceport":5432,"targetname":"hostname1","targetport":5432}
+(3 rows)
+
+-- manually balance the cluster such that we have
+-- a balanced cluster like above with 1,2,3,4,5,6 and hostname1/2/3
+-- shardid 85674049 (1) nodeid 58 (hostname1)
+-- shardid 85674050 (2) nodeid 58 (hostname1)
+-- shardid 85674051 (3) nodeid 59 (hostname2)
+-- shardid 85674052 (4) nodeid 59 (hostname2)
+-- shardid 85674053 (5) nodeid 60 (hostname3)
+-- shardid 85674054 (6) nodeid 60 (hostname3)
```

Comment:
nit: for simplicity of following the test case I think it would  be good to make the final digits match. So shift shardid by 2 and nodeids by 3

so you start with shardid 85674051 and nodeid 61